### PR TITLE
Improve and add packets/structs related to nearby info requests

### PR DESCRIPTION
**Pull request by cirras**

### CODE REVIEW

File: eo.txt [hunk 1]
```diff
@@ -489,6 +489,8 @@ struct CharacterMapInfo
 	struct Paperdoll_B000A0HSW paperdoll
 	SitState sit_state
 	char invisible
+	// TODO: Represent optionally present data
+	WarpAnimation animation
```
Comment (sorokya) [inline eo.txt:500]: nice, yeah I have this in reoserv too

File: eo.txt [hunk 2]
```diff
@@ -1394,9 +1396,13 @@ server_packet(Avatar, Remove)
 "Player has appeared in nearby view"
 server_packet(Players, Agree)
```
Comment (sorokya) [inline eo.txt:1422]: Does this really work? I'm pretty sure I captured the data from GameServer and the `break` and `1` are present.

File: eo.txt [hunk 3]
```diff
@@ -1394,9 +1396,13 @@ server_packet(Avatar, Remove)
...
 {
-	break
-	struct CharacterMapInfo character
-	char = 1
+	struct NearbyInfo nearby
+}
+
+"NPC has appeared in nearby view"
+server_packet(Appear, Reply)
```
Comment (sorokya) [inline eo.txt]: `Appear` has been renamed to `MapInfo` it's packet definition is
```
"MapInfo for characters and/or npcs"
server_packet(MapInfo, Reply)
{
	char num_entities
	break // omitted if not sending back characters
	struct CharacterMapInfo characters[num_entities] // omitted if not sending back characters
	struct NPCMapInfo npcs[] // reads to end
}
```

**Review by sorokya — CHANGES_REQUESTED** [review]
Some questions on Players_Agree. Can remove Appear_Reply as it's already represented with MapInfo_Reply

File: eo.txt (hunk 3 above)
Comment (cirras) [reply eo.txt]: Thanks, I'll move this change into `MapInfo, Reply` then.

File: eo.txt (hunk 2 above)
Comment (cirras) [reply eo.txt:1422]: The client expects a `NearbyInfo` struct and reads the garbage data send by GameServer due to a fluke of the weird packet-chunking behavior in the official packet reader.
Relevant conversation on this subject:
image
image
image
image
image
image

Comment (cirras) [reply eo.txt:1422]: More relevant conversation:
image
image

File: eo.txt [hunk 4]
```diff
@@ -2407,12 +2408,10 @@ client_packet(NPCMapInfo, Request)
 "MapInfo for characters and/or npcs"
 server_packet(MapInfo, Reply)
 {
-	char num_entities
-	break // omitted if not sending back characters
-	struct CharacterMapInfo characters[num_entities] // omitted if not sending back characters
-	struct NPCMapInfo npcs[] // reads to end
+	struct NearbyInfo nearby
```
Comment (sorokya) [inline eo.txt:2483]: Not sure I agree with this.. The packet structure is identical but sending ItemMapInfo structs doesn't actually do anything in the client. The client seems to just ignore them

Comment (sorokya) [reply eo.txt:2483]: I guess it's fine to use NearbyInfo for this. Maybe we just need a comment to say "Items are ignored"

Comment (cirras) [reply eo.txt:2483]: Hmm. Are you certain that the items are ignored?
Sausage seemed pretty convinced that the entirety of the packet data from both `PLAYERS_AGREE` and `APPEAR_REPLY` gets passed to the same `Handle_NearbyInfo` function in the client.

Comment (sorokya) [reply eo.txt:2483]: My bad - ItemMapInfo should use TinyCoords not Coords. It does work

**Pull request merged**